packages/lib/src/interactions/SelectionRect.tsx

```diff
@@ -17,17 +19,29 @@ function SelectionRect(props: Props) {
     endPoint,
     fill = 'red',
     fillOpacity = 0.5,
+    clampToVis,
     ...restSvgProps
   } = props;
 
   const { width, height } = useThree((state) => state.size);
 
   const htmlSelection = useCameraState(
-    (...args) => ({
-      startPoint: dataToHtml(...args, startPoint),
-      endPoint: dataToHtml(...args, endPoint),
-    }),
-    [startPoint, endPoint]
+    (camera, context) => {
+      const { visSize, dataToWorld, worldToData } = context;
+      const [newStartPoint, newEndPoint] = clampToVis
+        ? clampRectangleToVis(
+            dataToWorld(startPoint),
+            dataToWorld(endPoint),
+            visSize
+          )
+        : [dataToWorld(startPoint), dataToWorld(endPoint)];
```

Comment:
A bit fiddly here as we need the points in world coordinates to clamp to the vis :shrug:

Reply:
What about an early return for `!clampToVis` to avoid the `dataToWorld`/`worldToData` conversions?